### DISCUSSION TITLE
Want to write netCDF to tiff but cannot identify the `crs`?

### DISCUSSION THREAD

**Discussion by qAp** [corteva/rioxarray#229] [Q&A]
Hi
I'm trying to convert this netCDF file:
http://thredds-gfnl.usc.es/thredds/fileServer/GLOBALWTDFTP/annualmeans/NAMERICA_WTD_annualmean.nc
to a tiff file using `rioxarray` and to eventually upload the tiff onto the Google Earth Engine, which requires the tiff file to have a projection.  

I know from [Converting NetCDF dataset array to GeoTiff using rasterio Python](https://gis.stackexchange.com/questions/323317/converting-netcdf-dataset-array-to-geotiff-using-rasterio-python/329141#329141), that the projection can be set using something like `xds.rio.set_crs("epsg:4326")`

I have also found, following [Search order for the CRS (DataArray and Dataset)](https://corteva.github.io/rioxarray/stable/getting_started/crs_management.html#Search-order-for-the-CRS-(DataArray-and-Dataset):), that for my dataset `ds`, `ds['spatial_ref'].attrs['GeoTransform']` is:
```
-180.0000010006781 0.008333300184369964 0.0 83.99968819563394 0.0 -0.008333300447549506
```

But how do I use this info to set a projection before writing out to tiff?

**Comment by snowman2** [comment]
I recommend using:
```
xds.rio.write_crs("epsg:4326", inplace=True)
```
It stays around better.

**Reply by qAp** [reply]
Thanks.   I was able to set the projection successfully this way.  However, when writing it out to a tiff, it seems to hang indefinitely.  Was wondering if this is the right way to write out tiffs, from a array?  

```python
ds = rioxarray.open_rasterio(f'../data/{url.name}')
ds.rio.write_crs('epsg:4326', inplace=True)
print(ds.rio.crs)
ds = ds.squeeze()
ds.WTD.rio.to_raster(f'../data/NAMERICA_WTD_annualmean.tiff')
```

**Reply by qAp** [reply]
@snowman2 Apologies, it did write, just had to wait ~ 25 minutes.  If this is unusual, please let me know.   Thanks

**Reply by snowman2** [reply]
@qAp, there are a couple of things to think about:
- Sometimes I have has better luck using `open_dataset` with netCDF files when reading in the data.
- You are reading it over the network, how long does it take to download?
- Looking into parallel reading/writing with rioxarray 0.2 [dask example](https://corteva.github.io/rioxarray/stable/examples/dask_read_write.html) & [read parallel](https://corteva.github.io/rioxarray/stable/examples/read-locks.html).

**Reply by RichardScottOZ** [reply]
Yes, also how big it is - if something is gigabytes plus it will take a bit of time.
